waifu2x.ts

```diff
@@ -276,17 +299,14 @@ export default class Waifu2x {
         const files = fs.readdirSync(dir)
         let fileMap = files.map((file) => `${dir}/${file}`).filter((f) => fs.lstatSync(f).isFile())
         const dirMap = files.map((file) => `${dir}/${file}`).filter((f) => fs.lstatSync(f).isDirectory())
-        for (let i = 0; i < dirMap.length; i++) {
-            const search = Waifu2x.recursiveSearch(dirMap[i])
-            fileMap = [...fileMap, ...search]
-        }
-        return fileMap
+        return fileMap.concat(dirMap.flatMap((dirEntry) => Waifu2x.recursiveSearch(dirEntry)))
     }
 
     public static upscaleImages = async (sourceFolder: string, destFolder?: string, options?: Waifu2xOptions, progress?: (current: number, total: number) => void | boolean) => {
-        if (!options) options = {}
+        options = {...options} || {}
         const files = fs.readdirSync(sourceFolder)
         if (sourceFolder.endsWith("/")) sourceFolder = sourceFolder.slice(0, -1)
+        //isn't this the same as recursiveSearch?

```

Comment:
To answer this comment, it should only search recursively when the recursive option is set to true. If its false, it will only use the files in the top directory (basically only apply recursiveSearch once).

Reply:
Yes i meant the lines below look the same as the recursive search function, should I use that instead?

Reply:
You can refactor it to use recursiveSearch here but you would need to add a parameter to make it only return the files in the top directory (and not search recursively) if the `options.recursive` is set to `false`. 

The new definition of the function can look like this:
`private static recursiveSearch = (dir: string, recursive?: boolean)`

If it is false, you can immediately return the fileMap.
`if (!recursive) return fileMap`